packages/flutter_tools/lib/src/commands/update_packages.dart

```diff
@@ -1321,8 +1321,19 @@ class PubspecDependency extends PubspecLine {
 
   /// This generates the entry for this dependency for the pubspec.yaml for the
   /// fake package that we'll use to get the version numbers figured out.
-  void describeForFakePubspec(StringBuffer dependencies, StringBuffer overrides, { bool useAnyVersion = true}) {
-    final String versionToUse = useAnyVersion || version.isEmpty ? 'any' : version;
+  void describeForFakePubspec(StringBuffer dependencies, StringBuffer overrides, { bool doUpgrade = true }) {
+    final String versionToUse;
+    // This should only happen when manually adding new dependencies; otherwise
+    // versions should always be pinned exactly
+    if (version.isEmpty || version == 'any') {
```

Comment:
The || is extra it falls under the last else

Reply:
I think if `doUpgrade` is true, and `version = 'any'`, then without the `||`, we could end up with `versionToUse = '>= any'"`. Not sure if `pub` would like that.

Reply:
Right, then the || is fine